File: backend/models/Labour.js
const mongoose = require("mongoose");

const labourSchema =
new mongoose.Schema({

  name: {

    type: String,

    required: true,

  },

  phone: {

    type: String,

    required: true,

  },

  dailyWage: {

    type: Number,

    required: true,

  },

  assignedSite: {

    type:
      mongoose.Schema.Types.ObjectId,

    ref: "Site",

  },

  joiningDate: {

    type: Date,

    default: Date.now,

  },

  siteHistory: [

    {

      site: {

        type:
          mongoose.Schema.Types.ObjectId,

        ref: "Site",

      },

      fromDate: {

        type: Date,

        default: Date.now,

      },

      toDate: {

        type: Date,

        default: null,

      },

      active: {

        type: Boolean,

        default: true,

      },

    },

  ],

});

module.exports = mongoose.model(
  "Labour",
  labourSchema
);

File: mobile_app/lib/models/labour.dart
class Labour {
  final String id;
  final String name;
  final String phone;
  final double dailyWage;
  final DateTime? joiningDate;

  Labour({
    required this.id,
    required this.name,
    required this.phone,
    required this.dailyWage,
    this.joiningDate,
  });

  factory Labour.fromJson(Map<String, dynamic> json) {
    return Labour(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      dailyWage: (json['dailyWage'] ?? 0.0).toDouble(),
      joiningDate: json['joiningDate'] != null 
          ? DateTime.tryParse(json['joiningDate']) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
      'dailyWage': dailyWage,
      if (joiningDate != null) 'joiningDate': joiningDate!.toIso8601String(),
    };
  }
}
